pkg/workers/leader_election_mgr_test.go

```diff
@@ -185,3 +187,365 @@ func TestLeaderElectionManager_acquireLeaderLease(t *testing.T) {
+				},
+			},
+			want: &LeaderElectionManager{
+				workers:                                []Worker{},
+				connectionFactory:                      &db.ConnectionFactory{},
+				tearDown:                               nil,
+				leaderElectionReconcilerRepeatInterval: 15 * time.Second,
+				leaderLeaseExpirationTime:              1 * time.Minute,
+				workerGrp:                              sync.WaitGroup{},
+			},
+		},
+	}
+	g := NewWithT(t)
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			g.Expect(NewLeaderElectionManager(tt.args.workers, tt.args.connectionFactory, tt.args.reconcilerConfig)).To(Equal(tt.want))
+		})
+	}
+}
+
+func TestLeaderElectionManager_start(t *testing.T) {
+	type fields struct {
+		workers                                []Worker
+		connectionFactory                      *db.ConnectionFactory
+		tearDown                               chan struct{}
+		leaderElectionReconcilerRepeatInterval time.Duration
+		leaderLeaseExpirationTime              time.Duration
+		workerGrp                              *sync.WaitGroup
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		setupFn func()
+	}{
+		{
+			name: "should start the worker if the worker is a leader and is not running",
+			fields: fields{
+				workers: []Worker{
+					&WorkerMock{
+						GetIDFunc: func() string {
+							return "01"
+						},
+						GetWorkerTypeFunc: func() string {
+							return "leader"
+						},
+						IsRunningFunc: func() bool {
+							return false
+						},
+						GetSyncGroupFunc: func() *sync.WaitGroup {
+							return &sync.WaitGroup{}
+						},
+						StartFunc: func() {},
+					},
+				},
+				connectionFactory:                      db.NewMockConnectionFactory(nil),
+				leaderElectionReconcilerRepeatInterval: 15 * time.Second,
+				leaderLeaseExpirationTime:              1 * time.Minute,
+				workerGrp:                              &sync.WaitGroup{},
+			},
+			setupFn: func() {
+				mockEntry := map[string]interface{}{
+					"leader":  "01",
+					"expires": time.Now().Add(time.Hour),
+				}
+				mocket.Catcher.Reset().
+					NewMock().
+					WithQuery("SELECT * FROM leader_leases").
+					WithReply([]map[string]interface{}{mockEntry})
+				mocket.Catcher.NewMock().WithExecException().WithQueryException()
+			},
+		},
+	}
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			tt.setupFn()
+			s := &LeaderElectionManager{
+				workers:                                tt.fields.workers,
+				connectionFactory:                      tt.fields.connectionFactory,
+				tearDown:                               tt.fields.tearDown,
+				leaderElectionReconcilerRepeatInterval: tt.fields.leaderElectionReconcilerRepeatInterval,
+				leaderLeaseExpirationTime:              tt.fields.leaderLeaseExpirationTime,
+				workerGrp:                              sync.WaitGroup{},
+			}
+			s.Start()
+		})
```

Comment:
There is no assertion here, is this desired? If so, why?

Reply:
Mainly because the method does not return anything to check, So I wasnt sure maybe I could check wantErr?

Reply:
Moreover, you could pass some `mock worker` instead of an empty array and check that the worker method are invoked with the correct params